packages/functional-tests/tests/settings/changeEmail.spec.ts

```diff
@@ -3,105 +3,118 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { test, expect } from '../../lib/fixtures/standard';
+import { SettingsPage } from '../../pages/settings';
+import { ChangePasswordPage } from '../../pages/settings/changePassword';
+import { SecondaryEmailPage } from '../../pages/settings/secondaryEmail';
 
 const AGE_21 = '21';
-let newEmail;
 
 test.describe('severity-1 #smoke', () => {
   test.describe('change primary email tests', () => {
-    test.beforeEach(
-      async ({ credentials, pages: { settings, secondaryEmail } }) => {
-        test.slow();
-        await settings.goto();
-        await settings.secondaryEmail.clickAdd();
-        newEmail = credentials.email.replace(/(\w+)/, '$1_alt');
-        await secondaryEmail.addAndVerify(newEmail);
-        await settings.waitForAlertBar();
-        await settings.secondaryEmail.clickMakePrimary();
-      }
-    );
+    test.beforeEach(async () => {
+      test.slow();
+    });
 
     test('change primary email and login', async ({
       credentials,
       page,
-      pages: { login, settings },
+      pages: { login, settings, secondaryEmail },
     }) => {
+      await settings.goto();
+
+      const newEmail = credentials.email.replace(/(\w+)/, '$1_alt');
+      await addSecondaryEmail(settings, secondaryEmail, newEmail);
+
       await settings.signOut();
 
       // Sign in with old primary email fails
       await login.setEmail(credentials.email);
       await page.locator('button[type=submit]').click();
       await login.setPassword(credentials.password);
       await page.locator('button[type=submit]').click();
-      expect(await login.signInError()).toContain(
-        'Primary account email required for sign-in'
-      );
+
+      await expect(
+        page.getByText('Primary account email required for sign-in')
+      ).toBeVisible();
 
       // Success signing in with New email
       await login.useDifferentAccountLink();
       await login.login(newEmail, credentials.password);
```

Comment:
Is this usually part of credentials cleanup (target.auth.destroy) or emails cleanup (target.auth.accountDestroy)?

Reply:
We are changing the email address for the account, but we aren't using the `emails` fixture. So when the cleanup function runs to delete accounts created during the test, it will look for the initial email but not the new one. I don't think the account destroy function will work when using the previous email?

Reply:
Okay! I think I've fixed this now. Every time we change the primary email, I update the credentials (see https://github.com/mozilla/fxa/pull/16671/commits/b829fcaf2cf510f7d1a2618ed7af18c562cab31b)